### DISCUSSION TITLE
Feature: defaultValue and value to allow primitive types

### DISCUSSION THREAD

**Discussion by gregg-cbs** [JedWatson/react-select#4651] [Ideas]
Hey,

I am voting for defaultValue and value to allow primitive types
```
<Select name="countryCode" defaultValue="ZA"
```

### my reasoning
1. It doesnt follow native select convention to pass an object as the value. Its not intuitive.
2. Developers dont store the select value in a database as an object, the value will be stored as a primitive.
3. If a developer wants to set the defaultValue from a stored value then they would need to filter the array of options to find the object and give that to the select - seems tedious for every developer who uses this to do that when you could do it once.
4. If developers want the chosen object they should use an event like 'onChange' to get it.

Its a great plugin, here it is in the wild:

![image](https://user-images.githubusercontent.com/57114335/119233032-6a82ae80-bb27-11eb-90e9-ab44dea5b0b8.png)

**Comment by davebrown1975** [comment]
Upvote on this,  I just discovered the behavior and was really left thinking why would someone do this......  it's not natural.  It should follow standard HTML semantics for the select control as much as possible and build on that.  So passing value should behave as expect,  perhaps introduce a defaultValueObject for those who want to rebel and break the rules.... :-)
